CMakeLists.txt

```diff
@@ -493,72 +497,119 @@ file(GLOB_RECURSE srcs_external CONFIGURE_DEPENDS
 add_library(cradle_external STATIC ${srcs_external} ${headers})
 target_sources(cradle_external PRIVATE ${generated_headers} ${generated_srcs})
-target_compile_definitions(
-    cradle_external PUBLIC ${CONAN_DEFINES} ${CONAN_COMPILE_DEFINITIONS})
-target_include_directories(cradle_external PUBLIC
-    include
-    ${generated_src_dirs}
-    src)
 target_link_libraries(cradle_external PUBLIC
     cradle_websocket)
 
 # The WebSocket server
 add_executable(websocket_server src/server.cpp)
-target_include_directories(websocket_server PRIVATE
-    ${generated_src_dirs}
-    src)
 target_link_libraries(websocket_server
     cradle_websocket
     cradle_plugins_inner
     Boost::program_options)
-add_custom_command(TARGET websocket_server POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:websocket_server> ${deploy_dir})
 
 # The rpclib server
 file(GLOB_RECURSE srcs_rpclib_server CONFIGURE_DEPENDS
     "${CMAKE_CURRENT_SOURCE_DIR}/src/cradle/rpclib/server/*.cpp")
 add_executable(rpclib_server ${srcs_rpclib_server})
-target_include_directories(rpclib_server PRIVATE
-    ${generated_src_dirs}
-    src)
-target_link_libraries(rpclib_server
+target_link_libraries(rpclib_server PRIVATE
     cradle_plugins_thinknode
     cradle_thinknode
     cradle_plugins_inner
     Boost::headers
     Boost::program_options
     cppcoro
     fmt::fmt
-    rpclib::rpclib
+    rpclib::rpc
     spdlog::spdlog)
-add_custom_command(TARGET rpclib_server POST_BUILD
-    COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:rpclib_server> ${deploy_dir})
+
+# Create deploy_dir.h for providing access to the rpclib server executable.
+add_custom_command(
+    OUTPUT ${deploy_dir_file}
+    COMMAND ${CMAKE_COMMAND}
+        -D input_dir=${CMAKE_CURRENT_SOURCE_DIR}/cmake
+        -D output_dir=${intgen_src_dir}/cradle
+        -D deployed_executable="$<TARGET_FILE:rpclib_server>"
+        -D compiler_id=${CMAKE_CXX_COMPILER_ID}
+        -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/deploy_dir.cmake
+    DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/cmake/deploy_dir.h.in)
 
 # Convenience target denoting all servers
 add_custom_target(servers)
 add_dependencies(servers websocket_server rpclib_server)
 
-# Retrieve the token info if it exists
-if (DEFINED ENV{CRADLE_THINKNODE_API_TOKEN})
-    set(API_TOKEN $ENV{CRADLE_THINKNODE_API_TOKEN})
-elseif (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.token")
-    file(READ "${CMAKE_CURRENT_SOURCE_DIR}/.token" API_TOKEN)
-else()
-    message(WARNING "Couldn't find API token.")
-    set(API_TOKEN "(missing)")
+if (CRADLE_ENABLE_TESTING)
+    # Retrieve the token info if it exists
+    if (DEFINED ENV{CRADLE_THINKNODE_API_TOKEN})
+        set(API_TOKEN $ENV{CRADLE_THINKNODE_API_TOKEN})
+    elseif (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.token")
+        file(READ "${CMAKE_CURRENT_SOURCE_DIR}/.token" API_TOKEN)
+    else()
+        message(WARNING "Couldn't find API token.")
+        set(API_TOKEN "(missing)")
```

Comment:
The parentheses in `(missing)` tend to cause build errors.

Reply:
Should it just be `missing` then? Unless I'm mistaken, this code wasn't really changed by this PR. It's just been nested inside an `if` block.

Reply:
Indeed not part of this pull request, but it's been bothering me every now and then. I'd prefer `missing`.